docs: align docs with `FeedbackDataset` refactor using `tab-set`

alvarobartt opened the pull request:
# Description

This PR adds the `tab-set` for the outdated code-blocks with the upcoming release of Argilla v1.14.0 so as to showcase both the previous and the new code-blocks for users that are still using Argilla v1.14.0 or lower.

Additionally, the style has been fixed in some documents while reviewing the outdated code-blocks, some file paths references have been fixed to point to the HTML files via relative paths, and some minor details.

Finally, a `warning` has also been included to let the users know that Argilla v1.14.0 won't work for the moment with the `ArgillaCallbackHandler` in `LangChain`. 

**Type of change**

- [X] Documentation update

alvarobartt committed:
feat: add `load_feedback_dataset_from_huggingface`

Add `argilla/oasst_response_quality` and `argilla/oasst_response_comparison` as additional `FeedbackDataset` datasets to be downloaded as part of `quickstart.Dockerfile`

alvarobartt committed:
style: fix word casing

alvarobartt committed:
fix: replace `installation/installation.md` with `quickstart_installation.ipynb`

alvarobartt committed:
fix(style): outdated URL and extra code-block

alvarobartt committed:
docs: add `warning` for `ArgillaCallbackHandler`

alvarobartt committed:
docs(chore): update `FeedbackDataset` Python ref

alvarobartt committed:
fix(docs): add `tab-set` and fix style

`tab-set` is used to support both Argilla v1.14.0 or higher, as well as lower versions of Argilla for `FeedbackDataset` related operations

gabrielmbmb commented on docs/_source/guides/llms/conceptual_guides/rm.md:
````diff
@@ -121,10 +120,24 @@ for record in prompts:
+
+:::{tab-item} Lower than Argilla 1.14.0
+```python
+# This publishes the dataset with the records into Argilla and turns the dataset objetct into a dataset in Argilla
````
```suggestion
# This publishes the dataset with its records to Argilla and returns the dataset in Argilla
```

gabrielmbmb replied:
```suggestion
# This publishes the dataset with its records to Argilla and turns the dataset object into a dataset in Argilla
```

nataliaElv commented on docs/_source/guides/llms/conceptual_guides/rm.md:
````diff
@@ -121,10 +120,24 @@ for record in prompts:
+:::{tab-item} Argilla 1.14.0 or higher
+```python
+# This publishes the dataset with the records into Argilla and returns the dataset in Argilla
+remote_dataset = dataset.push_to_argilla(name="my-dataset", workspace="my-workspace")
````
The variable `remote_dataset` is not used in the rest of the code snippets in the page, so maybe `dataset.push_to_argilla()` is enough in this case?

alvarobartt replied:
Even though it may not be used later, I think it's nice to show that they need to capture the return object, as the `warning` will be removed in the next release, and that affects users if they copy-paste the snippets pretending to use those further

gabrielmbmb replied:
I agree with Alvaro. Even if it's not used, it's good to have it there so the user knows that the methods returns something that can be used.

nataliaElv commented on docs/_source/guides/llms/conceptual_guides/sft.md:
````diff
@@ -118,10 +119,24 @@ records = [
+:::{tab-item} Argilla 1.14.0 or higher
+```python
+# This publishes the dataset with the records into Argilla and returns the dataset in Argilla
+remote_dataset = dataset.push_to_argilla(name="my-dataset", workspace="my-workspace")
````
Same as above `remote_dataset` is not used later. Maybe we can just `push_to_argilla()` in this case?

gabrielmbmb commented on docs/_source/guides/llms/conceptual_guides/sft.md:
````diff
@@ -118,10 +119,24 @@ records = [
-# This publishes the dataset and pushes the records into Argilla
+:::{tab-item} Argilla 1.14.0 or higher
+```python
+# This publishes the dataset with the records into Argilla and returns the dataset in Argilla
````
```suggestion
# This publishes the dataset with its records to Argilla and returns the dataset in Argilla
```

gabrielmbmb commented on docs/_source/guides/llms/conceptual_guides/sft.md:
````diff
@@ -118,10 +119,24 @@ records = [
+
+:::{tab-item} Lower than Argilla 1.14.0
+```python
+# This publishes the dataset with the records into Argilla and turns the dataset objetct into a dataset in Argilla
````
```suggestion
# This publishes the dataset with its records to Argilla and turns the dataset object into a dataset in Argilla
```

nataliaElv commented on docs/_source/guides/llms/practical_guides/create_dataset.md:
````diff
@@ -381,8 +394,19 @@ record = rg.FeedbackRecord(
+::::{tab-set}
+
+:::{tab-item} Argilla 1.14.0 or higher
+```python
+remote_dataset = dataset.push_to_argilla(name="my-dataset", workspace="my-workspace")
+```
````
```suggestion
```python
# push and save the remote dataset as a variable
# recommended if you are making updates later on (e.g. adding more records)
remote_dataset = dataset.push_to_argilla(name="my-dataset", workspace="my-workspace")

# or: push directly without saving the variable if you don't need to make changes
dataset.push_to_argilla(name="my-dataset", workspace="my-workspace")
```
```

nataliaElv commented on docs/_source/guides/llms/practical_guides/fine_tune.md:
````diff
@@ -434,10 +433,26 @@ record = rg.FeedbackRecord(
+
+:::{tab-item} Argilla 1.14.0 or higher
+```python
+remote_dataset = dataset.push_to_argilla(name="pre-training")
````
Same as with other snippets `remote_dataset` is not being used so maybe it's not necessary to save as a variable here. However, in this specific page the code snippet below seems to use the dataset, so this snippet could be integrated here keeping the `remote_dataset` variable.

nataliaElv commented on docs/_source/guides/llms/practical_guides/set_up_annotation_team.md:
````diff
@@ -92,14 +93,31 @@ for username, records in assignments.items():
+
+:::{tab-item} Argilla 1.14.0 or higher
+```python
+remote_dataset = dataset.push_to_argilla(name="my_dataset", workspace=workspace.name)
````
Same here. Do we really need to save the `remote_dataset` if it's not going to be used later on? 
Also, I think it doesn't make sense to take this line of code out of the for loop above. If the original code snippet works in both versions then I would keep it as it is

nataliaElv reviewed (commented):
I think that in some cases the tabs make things a bit more complicated than they are. We should be more explicit about when it's necessary to do `remote_dataset = dataset.push_to_argilla(...)` and when `dataset.push_to_argilla(...)` is good enough. If we change it everywhere with the option that saves the dataset as a variable it looks as if this is the only option. I have made some comments and suggestions concerning this issue in my review.

alvarobartt commented on docs/_source/guides/llms/conceptual_guides/rm.md:
````diff
@@ -121,10 +120,24 @@ for record in prompts:
-# This publishes the dataset and pushes the records into Argilla
+:::{tab-item} Argilla 1.14.0 or higher
+```python
+# This publishes the dataset with the records into Argilla and returns the dataset in Argilla
````
```suggestion
# This publishes the dataset with its records to Argilla and returns the dataset in Argilla
```

alvarobartt committed:
Apply suggestions from code review

Co-authored-by: Gabriel Martín Blázquez <gmartinbdev@gmail.com>

alvarobartt committed:
Apply suggestions from code review

Co-authored-by: Gabriel Martín Blázquez <gmartinbdev@gmail.com>

alvarobartt committed:
Apply suggestions from code review

Besides @nataliaelv suggestions, some other things have also been fixed related to wrong snippets, formatting issues, and wrong URLs

Co-authored-by: Natalia Elvira <natalia@argilla.io>
